homeassistant/components/unifi/device_tracker.py

```diff
@@ -250,16 +212,25 @@ def entity_registry_enabled_default(self):
     async def async_added_to_hass(self):
         """Subscribe to device events."""
         LOGGER.debug("New UniFi device tracker %s (%s)", self.name, self.device.mac)
+        self.device.register_callback(self.async_update_callback)
+        self.listeners.append(
+            async_dispatcher_connect(
+                self.hass, self.controller.signal_reachable, self.async_update_callback
+            )
+        )
 
-    async def async_update(self):
-        """Synchronize state with controller."""
-        await self.controller.request_update()
+    async def async_will_remove_from_hass(self) -> None:
+        """Disconnect device object when removed."""
+        self.device.remove_callback(self.async_update_callback)
+        for unsub_dispatcher in self.listeners:
+            unsub_dispatcher()
 
-        LOGGER.debug(
-            "Updating UniFi tracked device %s\n%s",
-            self.entity_id,
-            pformat(self.device.raw),
-        )
+    @callback
+    def async_update_callback(self):
+        """Update the sensor's state."""
+        LOGGER.debug("Updating UniFi tracked device %s", self.entity_id)
+
+        self.async_schedule_update_ha_state()
```

Comment:
Should I do this for Deconz and Axis as Well then?

Reply:
If they also use the dispatch helper to signal an update there's no benefit of using `self.async_schedule_update_ha_state` since a job has already been scheduled by the dispatch helper.

Ie, with `async_schedule_update_ha_state` here, sending the dispatch signal will schedule a job that schedules a job that updates entity state. By using `async_write_ha_state` the signal will just schedule a job that updates entity state.

Reply:
Ok. Then it maybe I shouldn't change it at all since this callback is also used by client/device class in aiounifi to get hass to update the entity?

Reply:
Yeah, if we want to keep the initial callback as short as possible, that may be wise.

It's good to think about this, that's why I brought it up. You should decide.